src/tests/suite/dfs_unit_test.c

```diff
@@ -1187,7 +1187,7 @@ dfs_test_mt_connect(void **state)
 	}
 	pthread_barrier_wait(&barrier);
 	for (i = 0; i < dfs_test_thread_nr; i++)
-		rc = pthread_join(dfs_test_tid[i], NULL);
+		dfs_test_rc[i] = pthread_join(dfs_test_tid[i], NULL);
```

Comment:
this sounds wrong. it overwrites the actual RC that is set in the thread function
dfs_test_connect_thread()

Reply:
so just do something like:
rc = pthread_join()
assert_int_equal(rc, 0);

Reply:
You are correct I misinterpreted this section of code. I'll make this fix and then push the fix.